spec/enumerables_spec.rb

```diff
@@ -0,0 +1,242 @@
+require './app'
+
+describe Enumerable do
+  let(:array) { [1, 2, 3] }
+  let(:block) { proc { |x| x * 2 } }
+  let(:inject_block) { proc { |prod, n| prod * n } }
+  let(:block_with_index) { proc { |num, i| puts "Num: #{num}, index: #{i}" } }
+  let(:range) { (1..5) }
+  let(:my_hash) { { first_name: 'somoye', second_name: 'nusret' } }
+  let(:hash_block) { proc { |key, value| puts "#{key} is #{value}" } }
+  let(:rg) { '/d/' }
+
+  describe '#my_each' do
```

Comment:
You should check if it iterates over all the elements in the enumerable.
You should check it doesn't mutate the original array.